### PR TITLE
Add custom font files support

### PR DESCRIPTION
**Pull request by naveen521kk**
## Motivation
ManimPango v0.1.7 allows adding fonts to search path, for windows.
This will introduce a  helper function which can be used for properly registering and unregistering a font file.
Also, this allows searching files in asset directories as done in other parts of code.

## Overview / Explanation for Changes
A simple example,
```py
from manim import *
class CustomFont(Scene):
    def construct(self):
        with register_font("Orbitron-VariableFont_wght.ttf"):
            a=Text("Hello",font="Orbitron")
        self.play(Write(a))
```
After downloading the font from https://fonts.google.com/download?family=Orbitron and placing it in the same folder as the script I get this video

https://user-images.githubusercontent.com/49693820/104809910-e46ff780-5816-11eb-990c-9b6274969430.mp4

## Oneline Summary of Changes

```
- Added support for custom font files (:pr:`939`)
```

## Testing Status
Works locally on Windows.

## Further Comments
Currently, this doesn't work macOS. until ManimPango supports it.

## Acknowledgements
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)

## Reviewer Checklist
- [ ] Newly added functions/classes are either private or have a docstring
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings
- [ ] The oneline summary has been included [in the wiki](https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release)

### CODE REVIEW

File: manim/mobject/svg/text_mobject.py [hunk 1]
```diff
@@ -1387,3 +1387,74 @@ def extract_color_tags(self):
...
+        The font file to add.
+
+    Examples
+    --------
+    Use ``with register_font(...)`` to add a font file to search
+    path.
+
+    .. code-block:: python
+       >>> with register_font(...):
+       ...     a=Text("Hello",font=...)
```
Comment (naveen521kk) [inline manim/mobject/svg/text_mobject.py:1427]: @behackl do you know how to skip doc test here?

Comment (behackl) [reply manim/mobject/svg/text_mobject.py:1427]: Don't write it using the Prompt syntax:

```
.. code-block::

    with register_font(...):
        a = Text("Hello", font=...)
```

I'd also like to see more specific arguments in both cases (e.g., `register_font(path/to/font_file.ttf)`, `font="Custom Font Name"`).

Comment (naveen521kk) [reply manim/mobject/svg/text_mobject.py]: done

File: manim/mobject/svg/text_mobject.py [hunk 2]
```diff
@@ -1387,3 +1387,75 @@ def extract_color_tags(self):
...
+
+    Examples
+    --------
+    Use ``with register_font(...)`` to add a font file to search
+    path.
+
+    .. code-block:: python
+
+        with register_font("path/to/font_file.ttf"):
+           a=Text("Hello",font="Custom Font Name")
```
Comment (leotrs) [inline manim/mobject/svg/text_mobject.py:1428]:
```suggestion
           a = Text("Hello", font="Custom Font Name")
```

File: manim/mobject/svg/text_mobject.py [hunk 3]
```diff
@@ -1387,3 +1387,75 @@ def extract_color_tags(self):
...
+    """Context Manager which adds a font
+    file to Pango's search path.
+    This searches for the font_file at various places.
+
+    The order it searches it described below.
+
+    1. Absolute path.
+    2. In ``assets/fonts`` folder.
+    3. In ``font/`` folder.
+    4. In the same directory.
```
Comment (leotrs) [inline manim/mobject/svg/text_mobject.py:1403]:
```suggestion
    """Temporarily add a font file to Pango's search path.
    
    This searches for the font_file at various places. The order it searches it described below.

    1. Absolute path.
    2. In ``assets/fonts`` folder.
    3. In ``font/`` folder.
    4. In the same directory.
```

File: manim/mobject/svg/text_mobject.py [hunk 4]
```diff
@@ -1387,3 +1387,75 @@ def extract_color_tags(self):
...
+
+    .. note ::
+
+        This method of adding font files also works with
+        :class:`CairoText`
+
+    .. important ::
+
+        This method isn't available for macOS. Using this
+        method on macOS will raise an :class:`AttributeError`.
```
Comment (leotrs) [inline manim/mobject/svg/text_mobject.py:1413]: For consistency: please change `.. note ::` into a `Notes\n-----` section.

Please add the AttributeError to the `Raises` section.

File: manim/mobject/svg/text_mobject.py [hunk 5]
```diff
@@ -1387,3 +1387,75 @@ def extract_color_tags(self):
...
+        raise FileNotFoundError(error)
+
+    try:
+        assert manimpango.register_font(str(file_path))
+        yield
+    finally:
+        if sys.platform.startswith("linux"):
+            manimpango.unregister_font()
+        else:
+            manimpango.unregister_font(str(file_path))
```
Comment (leotrs) [inline manim/mobject/svg/text_mobject.py:1461]: Nitpick (non-blocking): It sounds like this check could be made inside `unregister_font` instead, no?

Comment (naveen521kk) [reply manim/mobject/svg/text_mobject.py:1461]: Well yeah, this can be done in manimpango itself, but currently, this is how the API is.

**Review by leotrs — APPROVED** [review]

File: manim/mobject/svg/text_mobject.py (hunk 5 above)
Comment (naveen521kk) [reply manim/mobject/svg/text_mobject.py:1461]: @leotrs can you make an issue in the other repo so that it can be fixed?

Comment (leotrs) [reply manim/mobject/svg/text_mobject.py:1461]: Done

File: manim/mobject/svg/text_mobject.py [hunk 6]
```diff
@@ -1387,3 +1387,75 @@ def extract_color_tags(self):
...
+    input_folder = Path(config.input_file).parent.resolve()
+    possible_paths = [
+        Path(font_file),
+        input_folder / "assets/fonts" / font_file,
+        input_folder / "fonts" / font_file,
+        input_folder / font_file,
+    ]
+    for path in possible_paths:
+        path = path.resolve()
+        print(path)
```
Comment (behackl) [inline manim/mobject/svg/text_mobject.py:1445]:
```suggestion
```

**Review by behackl — APPROVED** [review]
LGTM!

I only added one small suggestion, there seems to be a superfluous print statement.

**Pull request merged**